runtime/pkg/middleware/usage.go

```diff
@@ -0,0 +1,67 @@
+package middleware
+
+import (
+	"context"
+	"time"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/rilldata/rill/runtime/pkg/emitter"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"google.golang.org/grpc"
+)
+
+func UsageStreamServerInterceptor() grpc.StreamServerInterceptor {
+	return func(
+		srv interface{},
+		ss grpc.ServerStream,
+		info *grpc.StreamServerInfo,
+		handler grpc.StreamHandler,
+	) error {
+		subject := auth.GetClaims(ss.Context()).Subject()
+		if subject == "" {
+			subject = "anonymous"
+		}
+
+		newCtx := emitter.ContextWithUsageDims(ss.Context(),
+			*emitter.String("user_id", subject),
+			*emitter.String("request_method", info.FullMethod),
+		)
```

Comment:
1. Override the existing `ctx` instead of having `newCtx` – ensures no bugs will happen by accidentally using the old `ctx`
2. Feels like the pointer dereferencing here should not be necessary – can any of the funcs be refactored to make them work better together